guide.py

```diff
@@ -0,0 +1,434 @@
+    })
+
+    if self.LAUNCH_PARAM in self.request.params:
+      template_data[self.LAUNCH_PARAM] = True
+    if self.INTENT_PARAM in self.request.params:
+      template_data[self.INTENT_PARAM] = True
+
+    self._add_common_template_values(template_data)
+
+    self.render(data=template_data, template_path=os.path.join(path + '.html'))
+
+  def post(self, path, feature_id, stage_id):
+    user = users.get_current_user()
+    if user is None or (user and not self._is_user_whitelisted(user)):
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    spec_link = self.__FullQualifyLink('spec_link')
+
+    explainer_links = self.request.get('explainer_links') or []
+    if explainer_links:
+      explainer_links = filter(bool, [x.strip() for x in re.split('\\r?\\n', explainer_links)])
+
+    bug_url = self.__FullQualifyLink('bug_url')
+    intent_to_implement_url = self.__FullQualifyLink('intent_to_implement_url')
+    origin_trial_feedback_url = self.__FullQualifyLink('origin_trial_feedback_url')
+
+    ff_views_link = self.__FullQualifyLink('ff_views_link')
+    ie_views_link = self.__FullQualifyLink('ie_views_link')
+    safari_views_link = self.__FullQualifyLink('safari_views_link')
+    web_dev_views_link = self.__FullQualifyLink('web_dev_views_link')
+
+    # Cast incoming milestones to ints.
+    shipped_milestone = self.__ToInt('shipped_milestone')
+    shipped_android_milestone = self.__ToInt('shipped_android_milestone')
+    shipped_ios_milestone = self.__ToInt('shipped_ios_milestone')
+    shipped_webview_milestone = self.__ToInt('shipped_webview_milestone')
+    shipped_opera_milestone = self.__ToInt('shipped_opera_milestone')
+    shipped_opera_android_milestone = self.__ToInt('shipped_opera_android_milestone')
+
+    owners = self.request.get('owner') or []
+    if owners:
+      owners = [db.Email(x.strip()) for x in owners.split(',')]
+
+    doc_links = self.request.get('doc_links') or []
+    if doc_links:
+      doc_links = filter(bool, [x.strip() for x in re.split('\\r?\\n', doc_links)])
+
+    sample_links = self.request.get('sample_links') or []
+    if sample_links:
+      sample_links = filter(bool, [x.strip() for x in re.split('\\r?\\n', sample_links)])
+
+    search_tags = self.request.get('search_tags') or []
+    if search_tags:
+      search_tags = filter(bool, [x.strip() for x in search_tags.split(',')])
+
+    blink_components = self.request.get('blink_components') or models.BlinkComponent.DEFAULT_COMPONENT
+    if blink_components:
+      blink_components = filter(bool, [x.strip() for x in blink_components.split(',')])
+
+    devrel = self.request.get('devrel') or []
+    if devrel:
+      devrel = [db.Email(x.strip()) for x in devrel.split(',')]
+
+    try:
+      intent_stage = int(self.request.get('intent_stage'))
+    except:
+      logging.error('Invalid intent_stage \'{}\'' \
+                    .format(self.request.get('intent_stage')))
```

Comment:
A little surprised that we throw in the middle of processing.  Seems like we'd want to aggregate errors to the end.

Reply:
That's just logging that line to the GAE logs.   In the old form, I believe that is a <select> field, so users would need to go out of their way to submit a value that does not parse, in which case it defaults to stage 1 (Prototype).  When I rework the form processing, I'll make a helper function so that all the int fields are parsed and validated consistently.